src/routes/transactions/transactions.service.ts

```diff
@@ -218,6 +224,72 @@ export class TransactionsService {
     };
   }
 
+  private adjustTransactionHistoryPage(
+    limit?: number,
+    offset?: number,
+  ): PaginationData {
+    if (offset !== undefined && offset > 0) {
+      return new PaginationData(
+        limit ? limit + 1 : PaginationData.DEFAULT_LIMIT + 1,
+        offset - 1,
+      );
+    } else {
+      return new PaginationData(
+        limit ? limit : PaginationData.DEFAULT_LIMIT,
+        PaginationData.DEFAULT_OFFSET,
+      );
+    }
+  }
+
+  async getTransactionHistory(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    timezoneOffset?: string,
+    paginationData?: PaginationData,
+  ): Promise<Partial<TransactionItemPage>> {
```

Comment:
Why does this function return a `Partial` type?

Reply:
Because count is not optional and this endpoint is not returning the count field.

Reply:
But this might be related with the `Page` entity itself. I don't think returning a `Partial` type to get around the fact that it has `count` is the way to go. We are adding technical debt this way.

The issue is that we should not have any `count` at all when serialising the `Page` to the clients. This should be solved separately and not by adding `Partial` to this route.

Reply:
Ok, I removed the Partial and add count. 
0dac6dc15d88b563a60ab7a8d7d8e6367360f35f